bin/node/runtime/src/lib.rs

```diff
@@ -1603,30 +1603,38 @@ impl_runtime_apis! {
 		Block,
 		mmr::Hash,
 	> for Runtime {
-		fn generate_proof(leaf_index: pallet_mmr::primitives::LeafIndex)
-			-> Result<(mmr::EncodableOpaqueLeaf, mmr::Proof<mmr::Hash>), mmr::Error>
+		fn generate_proof(leaf_indexes: Vec<pallet_mmr::primitives::LeafIndex>)
+			-> Result<(Vec<(pallet_mmr::primitives::LeafIndex, mmr::EncodableOpaqueLeaf)>, mmr::Proof<mmr::Hash>), mmr::Error>
 		{
-			Mmr::generate_proof(leaf_index)
-				.map(|(leaf, proof)| (mmr::EncodableOpaqueLeaf::from_leaf(&leaf), proof))
+			Mmr::generate_proof(leaf_indexes)
+				.map(|(leaves, proof)| (leaves.iter().map(|l| (l.0, mmr::EncodableOpaqueLeaf::from_leaf(&l.1))).collect(), proof))
 		}
 
-		fn verify_proof(leaf: mmr::EncodableOpaqueLeaf, proof: mmr::Proof<mmr::Hash>)
+		fn verify_proof(leaves: Vec<mmr::EncodableOpaqueLeaf>, proof: mmr::Proof<mmr::Hash>)
 			-> Result<(), mmr::Error>
 		{
-			let leaf: mmr::Leaf = leaf
-				.into_opaque_leaf()
-				.try_decode()
-				.ok_or(mmr::Error::Verify)?;
-			Mmr::verify_leaf(leaf, proof)
+			let leaves = {
+				let mut decoded_leaves = Vec::new();
+				for l in &leaves {
+					let leaf: mmr::Leaf  = l.clone()
+						.into_opaque_leaf()
+						.try_decode()
+						.ok_or(mmr::Error::Verify)?;
+					decoded_leaves.push(leaf);
+				}
+				decoded_leaves
+			};
```

Comment:
rust has a cool feature for this

```suggestion
			let leaves = leaves.into_iter()
			    .map(|l| {
			        l.into_opaque_leaf()
						.try_decode()
						.ok_or(mmr::Error::Verify)
			})
			.collect::<Result<Vec<mmr::Leaf>, mmr::Error>>()?;
```

ru